software/winrm/src/main/java/org/apache/brooklyn/location/winrm/WinRmMachineLocation.java

```diff
@@ -197,7 +198,15 @@ public String getHostname() {
     @Nullable
     protected String getHostAndPort() {
         String host = getHostname();
-        return (host == null) ? null : host + ":" + config().get(WINRM_PORT);
+        return (host == null) ? null : host + ":" + (config().get(USE_SECURE_WINRM) ? 5986 : 5985);
+    }
+
+    public int getPort() {
+        if (getConfig(WINRM_PORT) != null) {
+            return getConfig(WINRM_PORT);
+        } else {
+            return getConfig(USE_SECURE_WINRM) ? 5986 : 5985;
```

Comment:
The config key duplication comes from the location deciding which login port to use, but the entity configuring the inbound ports. Seems there's no way for the location to influence the inbound ports. For ssh this works because 22 is the default for [`REQUIRED_OPEN_LOGIN_PORTS`](https://github.com/apache/brooklyn-server/blob/master/software/base/src/main/java/org/apache/brooklyn/entity/software/base/SoftwareProcess.java#L66). 

Getting the port from the location (above comment) will allow configuring the port at a single place.  

Cleaner solution will be to allow the location to add inbound ports and remove the keys from `VanillaWindowsProcessImpl`, but not in scope for this PR.